lib/assets/javascripts/cartodb/models/filter.js

```diff
@@ -587,7 +591,8 @@ cdb.admin.models.Filters = Backbone.Collection.extend({
       return f.getSQLCondition();
     })
 
-    var sql = _(sqls).compact().join(' AND ');
+    var operator = this.dataLayer.get('sql_or_toggle') ? ' OR ' : ' AND ';
```

Comment:
After looking at this again, it obfusticates things a bit, for no real gain atm. If this feature is revisited/expanded this code can be updated accordingly.